crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -177,18 +190,119 @@ impl<'a> Visitor<'a> {
 
             tasks.push(tokio::spawn(async move {
+                let task_id = info;
                 let _task_cache = task_cache;
                 let mut prefixed_ui =
-                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix);
-                prefixed_ui.output(command);
+                    Self::prefixed_ui(ui, is_github_actions, &output_client, pretty_prefix.clone());
+
+                let mut cmd = Command::new(package_manager.to_string());
+                cmd.args(["run", task_id.task()]);
+                cmd.current_dir(workspace_directory.as_path());
+                cmd.stdout(Stdio::piped());
+                cmd.stderr(Stdio::piped());
+
+                // TODO: avoid passing Duration::MAX and instead allow processes to run without
+                // a timeout
+                let mut process = match manager.spawn(cmd, Duration::MAX) {
+                    Some(Ok(child)) => child,
+                    // Turbo was unable to spawn a process
+                    Some(Err(e)) => {
+                        // Note: we actually failed to spawn, but this matches the Go output
+                        prefixed_ui.error(format!("command finished with error: {e}"));
+                        errors
+                            .lock()
+                            .expect("lock poisoned")
+                            .push(TaskError::from_spawn(task_id_for_display.clone(), e));
+                        callback
+                            .send(if continue_on_error {
+                                Ok(())
+                            } else {
+                                Err(StopExecution)
+                            })
+                            .ok();
+                        return;
+                    }
+                    // Turbo is shutting down
+                    None => {
+                        callback.send(Ok(())).ok();
+                        return;
+                    }
+                };
+
+                let exit_status = match process
+                    .wait_with_piped_outputs(
+                        PrefixedWriter::new(ui, pretty_prefix.clone(), output_client.stdout()),
+                        PrefixedWriter::new(ui, pretty_prefix.clone(), output_client.stderr()),
+                    )
+                    .await
+                {
+                    Ok(Some(exit_status)) => exit_status,
+                    Err(e) => {
+                        error!("unable to pipe outputs from command: {e}");
+                        callback.send(Err(StopExecution)).ok();
+                        manager.stop().await;
+                        return;
+                    }
+                    Ok(None) => {
+                        // TODO: how can this happen? we only update the
+                        // exit status with Some and it is only initialized with
+                        // None. Is it still running?
```

Comment:
@arlyon Do you know if this can happen and if so what it means? I followed through the code and I don't think this is a possible return value, so I'm not sure how to handle it.

Reply:
Yes, looks like you are correct. There was a time where if a process was killed by a signal we would mark that with a None exit code but since then we now return this ChildExit enum instead so the option is no longer needed. I can open a PR or you can handle it here up to you.

Reply:
I'm more than happy to handle in a follow up. Just wanted to double check that I wasn't missing some behavior.